### ISSUE TITLE
Question: Group by State in Location Sidebar

### ISSUE THREAD

**Issue by whreed** [bullhorn/career-portal#130]
Instead of displaying all the locations can I would like to just have the locations grouped by state.  Another Idea may be to have it grouped by state first (screenshot mock up below)....If this is not possible how can I go about changing the location sort to just by state and not city,state?  Client expectations are that they want it to work this way but I find it counter productive :)  

## Current behavior

-

## Expected behavior

-

## Steps to replicate behavior

1.

## Screenshots
![mockup](https://cloud.githubusercontent.com/assets/3143765/25319563/714aa796-286d-11e7-92f7-0574a21b3a3e.png)

**Comment by bulbajackel** [comment]
Hey whreed,

There was a previous discussion on this topic in #46. This may be what you are looking for if you were looking for the state alone.

For this more "accordion" approach, you will need to filter the city jobs by the states selected and from that list you will need to get the cities from your current job list.  It does appear as if you may need to extend the [search service](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js) for this functionality to group by state, instead of city and state.  Another area you will want to extend is the [controller for the sidebar](https://github.com/bullhorn/career-portal/blob/master/src/app/sidebar/sidebar.controller.js) as well. 

The data returned from a query should not be too hard to work with for this as it returns a value for city and a value for state in two separate keys.  I have included some sample data below for how your job JSON may look. 
```
{
    "start": 0,
    "count": 20,
    "data": [{
                "id": 128619,
                "title": "Applications Engineer",
                "employmentType": "Contract",
                "address": {
                    "state": "Massachusetts",
                    "city": "Somerville"
                },
                "dateLastPublished": 1492789927483,
                "isPublic": 1,
                "isOpen": true
            }
````



If you need some additional details for where to get started krsween has created [a nice writeup](https://github.com/bullhorn/career-portal/issues/58#issuecomment-219424529) for where to get started with modifying the filters.

Please let me know if you have any questions.

TL;DR: Yes this is possible, but it would require modification of the search service and the sidebar controller.

**Comment by whreed** [comment]
Thank you so much for that detailed response.  I have a client that I did a public facing site and now wants this integration into Bullhorn and linked me to this github career portal.  Is this extending something already created by Bullhorn?  I don't have access to Bullhorn and keep going through my client to get them to talk to support, so I am really green on this and even very green with AngularJS.  I did pull down this github and get it working with the source and being able to build locally to see my changes.  My other question would be in that topic #46 you linked, where do I go to edit a query?  I may have missed it in the source here....Also is this the place to drop questions like this or should I be asking somewhere else?

**Comment by bulbajackel** [comment]
Heya,
The [Bullhorn Career Portal](https://github.com/bullhorn/career-portal) integrates with [The Bullhorn ATS/CRM](http://www.bullhorn.com/).  Working with the portal doesn't require access to their Bullhorn instance itself, but it can simplify the testing process as you can publish test jobs, this is something that you can request from the client you are working with however.  

Each query in the app uses a prefix of `https://public-rest[Swimlane].bullhornstaffing.com/rest-services/[corp token].` You will see this referenced in the [search service](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js) as the variable [publicServiceUrl ](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js#L87).  This will not need to be touched.

That [search service file](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js) is where you modify the queries used for filtering.  The key area where you will be modifying the query will be located [here ](https://github.com/bullhorn/career-portal/blob/master/src/app/services/search.service.js#L126-L150) as it is where you will be able to modify the request parameter for location. 

In the sidebar, you will want to adjust the logic that pushes the city + state to the aformentioned function [here ](https://github.com/bullhorn/career-portal/blob/master/src/app/sidebar/sidebar.controller.js#L170-L179)

I know I did a bit of "you go here and here" in that post so please let me know if you need any assistance with anything please let me know.  I tried to make sure I highlighted ranges to point out the start and ends of a couple of the essential functions. 

Please let me know if you have any more questions.
